servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -331,6 +329,25 @@ public Completable closeAsyncGracefully() {
         return asyncCloseable.closeAsyncGracefully();
     }
 
+    /**
+     * {@link LoadBalancerFactory} for {@link RoundRobinLoadBalancer}.
+     *
+     * @param <ResolvedAddress> The resolved address type.
+     * @param <C> The type of connection.
+     */
+    public static final class RoundRobinLoadBalancerFactory<ResolvedAddress, C extends LoadBalancedConnection>
```

Comment:
can you clarify why this needs to be exposed as public type and have a static factory method to create a new one?

Reply:
The complexity here is that we want to make sure a `LoadBalancerFactory` can specify that it does not block (implement `HttpExecutionStrategyInfluencer`) but `LoadBalancerFactory` is protocol agnostic hence it can not implement that. 

Now, users can specify `RoundRobinLoadBalancer` factory to the client without realizing that they need to wrap with `HttpExecutionStrategyInfluencer`. This is an attempt to identify that the passed `LoadBalancerFactory` is provided by ST and hence we can make some assumptions.

Alternatively, we can say that if a user provides any LB factory (even if it is RRLB) we will offload and they should make sure to wrap, it seemed we can do better than that.